### ISSUE TITLE
Spurious comparison differences in workflows 136.X, 141.044, 4.54, ...

### ISSUE THREAD

**Issue by perrotta** [cms-sw/cmssw#43456]
Since a few IB's the workflows in the title show spurious comparison differences in DQM tests.
There are 6 failing comparisons in ParticleFlow_JetResponse_slimmedJets and 6 failing comparisons in ParticleFlow_JetResponse_slimmedJetsPuppi, both in the JEC subfolder.

Examples in:
- #43444 , baseline CMSSW_14_0_X_2023-11-30-1100
- #43451 , baseline CMSSW_14_0_X_2023-11-30-1100
- #43447 , baseline CMSSW_14_0_X_2023-11-30-1100
- #43445 , baseline CMSSW_14_0_X_2023-11-30-0800
- #43442 , baseline CMSSW_14_0_X_2023-11-29-1100
- ...
 
Well, all PRs I can see that were run with the latest IB's show this "feature", while I don't see it for PRs tested with CMSSW_14_0_X_2023-11-28-1100 (e.g.)

The PRs that were merged between CMSSW_14_0_X_2023-11-28-1100 and CMSSW_14_0_X_2023-11-29-1100 are:
[#43404](https://github.com/cms-sw/cmssw/pull/43404) from smuzaffar: [SIMULATION] [DEVEL] Fix warnings on deprecated copy-constructors
[#43338](https://github.com/cms-sw/cmssw/pull/43338) from fwyzard: Extend the `GenericConsumer` to consume individual products
[#43403](https://github.com/cms-sw/cmssw/pull/43403) from civanch: [14_0_X SIM] Update FTFP_BERT_EMN physics
[#43339](https://github.com/cms-sw/cmssw/pull/43339) from bouchamaouihichem: Enabling the JetCore+DeepCore2.2.1 hybrid with MaxCand=30 (JC30DC)
[#43242](https://github.com/cms-sw/cmssw/pull/43242) from francescobrivio: Read VertexSmearing from GT in Run-1/2/3 MC
[#42936](https://github.com/cms-sw/cmssw/pull/42936) from fwyzard: Fix logic error in `FinalPath` checks
[#43164](https://github.com/cms-sw/cmssw/pull/43164) from RSalvatico: Change EGM code to be able to read HCAL PF thresholds from DB
[#43386](https://github.com/cms-sw/cmssw/pull/43386) from mmusich: [TkAl] remove "old" all-in-one metatool from cmssw
[#43385](https://github.com/cms-sw/cmssw/pull/43385) from mmusich: fix HLT track collection for SiStrip at HLT monitoring
[#43348](https://github.com/cms-sw/cmssw/pull/43348) from ggonzr: Update `hash_name` function for RelMon utilities
[#43176](https://github.com/cms-sw/cmssw/pull/43176) from swagata87: Migrate RecHit-based rho producer to use HCal noise cuts from GT
[#43410](https://github.com/cms-sw/cmssw/pull/43410) from aandvalenzuela: [ALCA][DB] Fix copy-constructor warnings reported in DEVEL IBs
[#43304](https://github.com/cms-sw/cmssw/pull/43304) from peteryouBuffalo: UPDATED: Adding of jet purity and efficiency plots to PFValidation package

The last PR (#43304) at a first glance could be the responsible...,

**Comment by perrotta** [comment]
assign pf, dqm

**Comment by makortel** [comment]
@civanch Why assing to `simulation`? On a quick look only(? or mainly?) data workflows are affected.

**Comment by mmusich** [comment]
> The last PR (https://github.com/cms-sw/cmssw/pull/43304) at a first glance could be the responsible...,

What about https://github.com/cms-sw/cmssw/pull/43304#discussion_r1412851955 ?

**Comment by perrotta** [comment]
@peteryouBuffalo in https://github.com/cms-sw/cmssw/pull/43304#pullrequestreview-1760937740 was pointed out that there is an issue with the case `if nEvents == 0`.
In that thread you wrote "I will try to reach out and discuss with my colleagues at Buffalo who originally started and helped with the original PR (https://github.com/cms-sw/cmssw/pull/39902), so we can figure out resolving this specific case. My assumption is that it should be fine to just `continue` if` nEvents == 0`, but I will check with others to confirm." (which I report here for completeness).

This issue is becoming annoying, and it can make difficult to check and review new PR. Did you verify whether it is correct to just `continue` after [L126](https://github.com/cms-sw/cmssw/pull/43304/files/0adf85bb51dcd319915f81d5738e67a301a0381a#diff-24deb2f8761a760594e61245105ec302e7436a4811a0e1bfb512e31865101ab3R126) if nEvents = 0? At the first order, I would just verify that it is not disrupting anything, as it make sense not continuing there if there are no events.

Could you please let us know if a PR is coming? Otherwise we can just try that suggestion in a fix PR and see what happens...

**Comment by peteryouBuffalo** [comment]
@perrotta We have a PR coming to fix this issue. @laurenhay is going to make a PR to add a statement to fix the `nEvents == 0` case. We will have the PR taken care of as soon as possible.

**Comment by perrotta** [comment]
#43530 fixed the issue of the weird behaviour when `nEvents = 0`

It looks like it also fixed what pointed out here, that is the spurious differences in the test outputs of several workflows. 

In fact, the PRs tested after #43530 was merged (since CMSSW_14_0_X_2023-12-11-2300) do not show those differences in the test output any more, see for example [#43544](https://github.com/cms-sw/cmssw/pull/43544#issuecomment-1851608757) or [#42633](https://github.com/cms-sw/cmssw/pull/42633#issuecomment-1852260012)

Therefore, I think that this issue can get closed now
